buildkitd/certificates.go

```diff
@@ -247,3 +249,58 @@ func savePEM(path, typ string, bytes []byte) error {
 
 	return nil
 }
+
+// ConfigureSatelliteTLS uses the CA cert and key associate with the satellite
+// to generate a new certificate/key pair for use in client-side mTLS.
+// The certificates are configured in the settings for a new buildkit client.
+// The certificates are also stored on disk in: `~/.earthly/certs/<org-id>/<sat-name>`.
+func ConfigureSatelliteTLS(sat *cloud.SatelliteInstance, settings *Settings, cfg *config.Config) error {
+	dir := filepath.Join(cfg.Global.SatelliteCertsDir, sat.Org, sat.Name)
```

Comment:
using the same directory (as specified by a config file) makes this code susceptible to race conditions (where two instances of earthly are running in parallel).

can we use a temp directory instead?